internal/cmd/providers.go

```diff
@@ -36,100 +37,166 @@ func providersCmd() *cobra.Command {
 
 func createProviderCmd() *cobra.Command {
 	var (
-		hostname       string
-		clusterAddress string
+		fromContext   bool
+		clusterCA     string
+		clusterAddr   string
+		SAToken       string
+		namespace     string
+		storageClass  string
+		sshEnabled    bool
+		environmentSA string
+		orgAllowlist  []string
 	)
 	cmd := &cobra.Command{
-		Use:   "create [name] --hostname=[hostname] --cluster-address=[clusterAddress]",
+		Use:   "create [name]",
 		Args:  xcobra.ExactArgs(1),
 		Short: "create a new workspace provider.",
 		Long:  "Create a new Coder workspace provider.",
-		Example: `# create a new workspace provider in a pending state
+		Example: `# create a new workspace provider
 
-coder providers create my-provider --hostname=https://provider.example.com --cluster-address=https://255.255.255.255`,
+coder providers create my-provider`,
 		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
Is it possible to add a "dry run" mode which can dump all the resulting YAML files? That way, someone can send the configs to their k8s admin to run, if they don't have the necessary cluster permissions. Kind of like `helm template` does